src/site/topics/queues/queue-adt.rst

```diff
@@ -3,123 +3,166 @@ The Queue ADT
 ================
 
-* Like other collections, we need ways to
-    * Add something to the collection
-    * Remove something from the collection
+* Like other collections, there should be a way to
+
+    * Add something to the queue
+    * Remove something from the queue
     * Look at something, but do not remove it
 
-* Within the context of a queue, we have
-    * **Enqueue**
-        * Add something to the rear/tail
-    * **Dequeue**
-        * Remove something from the front/head
-    * **First**
-        * Look at the thing at the front/head of the queue, but do not remove it
 
-* We will also want our ``isEmpty``, ``size``, and ``toString``
+* Within the context of a queue, these operations are called
+
+    * Enqueue
+
+        * Add an element to the collection
+        * The element added will be the new rear of the queue
+
+
+    * Dequeue
+
+        * Remove an element from the collection
+        * The removed element will be from the front of the queue
+        * The element after the removed element will be the new front, if it exists
+        * The element removed is returned
+
+
+    * First
+
+        * Return the element at the front of the queue, but leave it in the queue
+        * Calling first does not alter the queue
+
+
+.. note::
+
+    It is against this definition of a queue to access anything from anywhere other than the *front* of the queue.
+
 
 .. warning::
 
-    If you look at `Java's Queue Interface <https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/Queue.html>`_, you will
-    see that they use ``add``, ``remove``, and ``element`` along with ``offer``, ``poll``, and ``peek``. The first three
-    are the same as our ``enqueue``, ``dequeue``, and ``first`` and the other three are the same, but do not throw
+    When referring to Java's actual
+    `Queue Interface <https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Queue.html>`_, one may
+    notice ``add``, ``remove``, and ``element``, along with ``offer``, ``poll``, and ``peek``. The first three are the
+    same as the defined ``enqueue``, ``dequeue``, and ``first``. The other three are the same, but do not throw
     exceptions.
 
-    We will use the ``enqueue``, ``dequeue``, and ``first`` names as it is what they are `typically called when referring
-    to a Queue ADT <https://en.wikipedia.org/wiki/Queue_(abstract_data_type)>`_.
+    The names ``enqueue``, ``dequeue``, and ``first`` are used here as they are the typical names used when
+    `referring to a Queue ADT <https://en.wikipedia.org/wiki/Queue_(abstract_data_type)>`_.
+
 
 
-Example Uses
-============
+Example Use
+===========
 
-* Example
-    * Your typical queueing line, like at a grocery store checkout
+* Examples
+
+    * Any typical single file queueing line, like at a grocery store checkout
     * Cars at a stoplight
     * Input buffers
 
-* We can get a little more sophisticated
 
-Caesar Cipher
--------------
+Rotating Cipher
+---------------
```

Comment:
Reading my old comment, the realization that queues are basically:
* breadth first search
* decoupling GODs (any kind of inter-process communication is typically via event queues, which allows B to depend on A without A needing to invoke B in any way)

Neither of those are really good for introductory examples. I can't think of any better examples off the top of my head.

Reply:
When cleaning it earlier, the BFS was the big thing I could think of, but yearh, can't jump to that quite yet. Fortunately we see good uses (A* (queue++) and BFS).